src/frontend/js/hooks/useUnionResource/index.ts

```diff
@@ -87,6 +90,20 @@ const useUnionResource = <
   const eofRef = useRef<Record<string, number>>(queryClient.getQueryData(eofQueryKey) ?? {});
   log('eof', eofRef.current);
 
+  const reset = () => {
+    setStack([]);
+    setPage(0);
+    setTotalCount(undefined);
+    setError(undefined);
+    setCursor(perPage);
+    setIntegrityCount(0);
+  };
+
+  if (refetchOnInvalidation) {
+    useQueryKeyInvalidateListener(queryAConfig.queryKey, reset);
+    useQueryKeyInvalidateListener(queryBConfig.queryKey, reset);
+  }
```

Comment:
It's weird to have a conditional hook.
May be use a `enable` flag in `useQueryKeyInvalidateListener`

Reply:
Yes I get what you mean, the key point to keep in mind is that we cannot change this flag afterwards, or it will automatically trigger an error. I prefer having this pattern to avoid having to create listeners where we don't need them, for performance purpose.

Reply:
ok, let's say that no one will dynamically change the value of `refetchOnInvalidation`